### ISSUE TITLE
Characters aren't scanned.

### ISSUE THREAD

**Issue by princessGasmask** [otaniemi-triforce/M.A.T.C.H.#1]
Hey, sorry to bump this if it's dead but I have just about everything running on my machine. The only issue I'm having is that despite the character folder being defined and characters being in the select, the bot always returns that there are no characters found. I feel like there's something really stupid and simple that I'm overlooking, but any help would be appreciated.

**Comment by Yarb** [comment]
Thank you for pointing that out, as that one thing led several fixes that should make project more usable outside our own little experiments.
The system was looking for the select.def from a location that was unique to our setup. 
This has now been fixed and location parameter has been moved to the config file along with several other relevant parameters that control the behavior of the mugenoperator.
I also updated the instructions to be bit more specific regarding the setup process.

**Comment by princessGasmask** [comment]
With the new guide, I got the Title Screen setup to match and the bot connects to Discord properly. However, the characters folder still isn't scanned despite matching the config.

Edit to avoid double-posting: Is there a certain structure that the select.def has to have? At the moment, it's just a list of my characters and nothing else. The characters are read in-game fine, but not by mugenoperator.

**Comment by Yarb** [comment]
Yeah, select.def should contain the characters and optionally definitions for extrastages, music and suchlike. Mugenoperator should ignore those but to be on safe side, place characters before any other definitions (which I assume where Mugen wants to find them anyway).
While looking at this, I noticed that mugenoperator parser was assuming *nix-style path separators(/) and ignoring the Windows-style (\)...which I guess anyone would default to this being Windows software after all.
That should now be fixed as well and might very well sort out your issue.

**Comment by princessGasmask** [comment]
I tried with the new mugenoperator. Still no dice. Also my characters list is just a characters list with no extra stages or music.

**Comment by Yarb** [comment]
Can you post select.def that you are using or an example from it? Maybe that will shed some light on the issue.

**Comment by princessGasmask** [comment]
;---------------------------------------------------------------------
[Characters]
 ;How to add characters
 ;---------------------
 ;Use the format:
 ;    charname, stagefilename
 ;
 ;eg. For a player with filename "kfm",
 ;    and stage filename "stages/mybg.def"
 ;type (without the semicolon):
 ;    kfm, stages/mybg.def
 ;
 ;If you want to load a different def file, you can enter it
 ;as a directory plus the def file. This example loads up
 ;chars/kfm/alt-kfm.def:
 ;    kfm/alt-kfm.def, stages/mybg.def
 ;
 ;If you put "random" as the stagefilename, then a random stage will
 ;be selected for that player. eg.
 ;    kfm, random
 ;
 ;Zipped characters are also supported.
 ;Place the ZIP file in the chars/ directory. The syntax is as
 ;follows:
 ;    zipname.zip:defname.def, stages/mybg.def
 ;For example, if you have newsuave.def in suave.zip, the syntax is:
 ;    suave.zip:newsuave.def, stages/mybg.def
 ;If your def file has the same name as the zip file (eg. suave.def
 ;in suave.zip), you can just put the name of the zip file alone:
 ;    suave.zip, stages/mybg.def
 ;
 ;Optional parameters may follow on the same line, separated
 ;by commas. Each parameter has the form:
 ;    paramname = paramvalue
 ;where paramname is the name of the parameter type, and
 ;paramvalue is the value to assign that parameter.
 ;The optional parameters are:
 ;
 ; - music
 ;   Set the paramvalue to the name of the music file to use
 ;   as the BGM for that character. This overrides the bgmusic
 ;   parameter in the stage's .def file, so you can re-use the
 ;   same stage for multiple characters, but have a different
 ;   BGM playing for each person.
 ;
 ; - includestage
 ;   Set the paramvalue to 0 to avoid including this stage
 ;   in the stage select list (in VS, training modes, etc)
 ;
 ; - order
 ;   Set the paramvalue to the ordering priority to give the
 ;   character. Valid values are from 1 to 30. A smaller value
 ;   means you will fight the character sooner. You will never
 ;   fight an order 2 character before an order 1 character,
 ;   and never an order 3 character before an order 2 one.
 ;   For example, you might want to set your boss character
 ;   to have order=3. The default order value is 1 if you omit
 ;   this param. See *.maxmatches under [Options] for how to
 ;   limit the number of matches per order priority.
 ;
 ;   Some examples:
 ;        kfm, stages/mybg.def, includestage=0
 ;        kfm, stages/mybg.def, music=sound/song.mp3
 ;        kfm, stages/mybg.def, music=sound/song.mp3, includestage=0
 ;        kfm, stages/mybg.def, order=3
 ;
 ;You can also add a randomize icon to the select screen. To do
 ;this, put the word "randomselect" on a line of its own, with no
 ;extra parameters.
 ;
 ;Insert your characters below.

Kamehameha
123_RE
akfgz
Akiha
Azumi_Hojo
BlackRaiyaD4
brian-jmax-kofa
Cynthia
Farah
Isurugi_Remi
J_chen
Maori
mariko
Marowak
NaedokoYuri
RYUKO2nd
saki_mamiya_z_3_00
Samus
Sanae
senna
SvCVampiro
Toriel
Vampirella
VanessaXI


;-----------------------
[ExtraStages]
 ;Put extra stages here. They will be available in VS and Watch modes.
 ;For example, you can insert "stages/mybg.def".
stages/stage0-720.def

;---------------------------------------------------------------------
[Options]
 ;Here you set the maximum number of matches to fight before game ends
 ;in Arcade Mode. The first number is the number of matches against
 ;characters with order=1, followed by order=2 and order=3 respectively.
 ;For example, for 4,3,1 you will fight up to 4 randomly-picked
 ;characters who have order=1, followed by 3 with order=2 and 1 with
 ;order=3.
 ;Up to 30 entries allowed.
arcade.maxmatches = 6,1,1,0,0,0,0,0,0,0

 ;Maximum number of matches to fight before game ends in Team Mode.
 ;Like arcade.maxmatches, but applies to Team Battle.
team.maxmatches = 4,1,1,0,0,0,0,0,0,0

**Comment by princessGasmask** [comment]
That's the file in it's entirety. It's mostly stock-standard aside from the character list.

**Comment by Yarb** [comment]
Ok, just clarify. Are those characters in their individual folders within the chars folder?

**Comment by princessGasmask** [comment]
Yes, that's correct. The folder names also match the names they have in the list. There are also .def files within them matching the folder name.

**Comment by Yarb** [comment]
Yeah, that seems to be problem. The parser ignores that little quality of life feature and assumes that format `folder/char(.def)` is used. Mulusu is working on a fix to that.

Meanwhile, doing the following change:
char -> char/char   (Example:  Akiha -> Akiha/Akiha)

Should allow you to test the system at the very least.

**Comment by princessGasmask** [comment]
I had tried that as well since I saw that was how it was formatted in the example select.def from the MUGEN-tournament repo. It didn't work either, so I changed back to my original formatting. I'll try it again now that the mugenoperator was updated, though, and see how that goes.

EDIT: Tried again. Still no dice. I figure that it may help to give more context behind the way I'm running the scripts. So, when I try to run via python in Powershell/terminal, I get error message: "ImportError: DLL load failed while importing _socket: %1 is not a valid Win32 application."

So I decided to try running the project via Spyder instead and that works as far as I can tell. I can get to the point of the project launching...but now we have the issue where it can't seem to find my characters. I'm not sure if the difference in the way I'm running the project is causing the problem or if there is a way to just fix the original Import Error.
